src/native/managed/cdacreader/tests/LoaderTests.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.Diagnostics.DataContractReader.UnitTests;
+
+using MockLoader = MockDescriptors.Loader;
+
+public unsafe class LoaderTests
+{
+    private static void RunLoaderContractTest(TargetTestHelpers helpers, Func<MockMemorySpace.Builder, Dictionary<DataType, Target.TypeInfo>, MockMemorySpace.Builder> configure, Action<Target> testCase)
+    {
+        var types = MockLoader.Types(helpers);
+
+        MockMemorySpace.Builder builder = new(helpers);
+        builder = builder
+            .SetContracts([nameof(Contracts.Loader)])
+            .SetTypes(types);
+
+        if (configure != null)
+        {
+            builder = configure(builder, types);
+        }
+
+        bool success = builder.TryCreateTarget(out ContractDescriptorTarget? target);
+        Assert.True(success);
+        testCase(target);
+    }
```

Comment:
We don't really need the "configure"/"testCase" split anymore - it was an artifact of using `fixed` for the descriptor initially.   You can keep it, of course, if you find it helpful as an organizational aid or to remove boilerplate.